packages/crdt/src/crdt/timestamp.ts

```diff
@@ -64,8 +64,14 @@ export function deserializeClock(clock: string): Clock {
     };
   }
 
+  const ts = Timestamp.parse(data.timestamp);
+
+  if (!ts) {
+    throw new Timestamp.InvalidError(data.timestamp);
+  }
```

Comment:
We were missing a validity check here. `data.timestamp` _might_ sometimes be invalid and it would break further down the chain. Now we have a more tailored exception for this case.